### ISSUE TITLE
Navigating to pages do not include the required trailing slash

### ISSUE THREAD

**Issue by kylebshr** [apple/swift-docc#323]
Navigating to a page does not include a trailing slash, so copying the link and opening it directly doesn't work. 

For example, our docs are hosted at https://example.com/marketui/documentation/marketui/

If you click on a type to see its details, the URL is directed to https://example.com/marketui/documentation/marketui/marketactionsheetcontroller

Copying that link and pasting it in to a browser gives the following error, unless a trailing slash is added:

<img width="996" alt="Screen Shot 2022-06-27 at 11 11 21 AM" src="https://user-images.githubusercontent.com/3526783/176008075-b0981abe-0739-4d57-bfd3-f546a8bb7ea4.png">

Replace this paragraph with a short description of the incorrect behavior. 
If you think this issue has been recently introduced and did not occur in an 
earlier version, please note that. If possible, include the last version that 
the behavior was correct in addition to your current version.

### Checklist
- [x] If possible, I've reproduced the issue using the `main` branch of this package.
- [x] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/apple/swift-docc/issues).

### Expected behavior
Links in the documentation should work when copy-pasted.

### Actual behavior
Links in the documentation do not work when copy-pasted.

### Steps to Reproduce
See above.

### Swift-DocC Version Information

**Swift-DocC version:** Xcode 13.3.1
**Swift Compiler version:** `swift-driver version: 1.45.2 Apple Swift version 5.6 (swiftlang-5.6.0.323.62 clang-1316.0.20.8)`

**Comment by kylebshr** [comment]
Hmm interesting - I don't know a lot about the infrastructure, but the DocC files are in an S3 bucket. It seems like the S3 API is looking for a file with that name rather than detecting that it should serve the directory.

**Comment by mportiz08** [comment]
I think you're right. I'm not super familiar with S3 myself, but the [documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/IndexDocumentSupport.html#IndexDocumentsandFolders) makes it seem like the default behavior should work here by redirecting to the trailing slash style URL, but it's possible that I'm misreading it or overlooking some detail as to why that's not working in this case.

<img width="607" alt="Screen Shot 2022-06-27 at 4 53 58 PM" src="https://user-images.githubusercontent.com/212918/176058475-9cbfb796-a47e-4d5c-bc7d-74830356a1b8.png">

**Comment by jtbandes** [comment]
Different servers and hosting tools behave differently in this regard. For example, Google Cloud Storage buckets have a configuration option called [`MainPageSuffix`](https://cloud.google.com/storage/docs/static-website#specialty_pages) which can mimic this behavior. However, the MainPageSuffix **only works when accessing the bucket through a custom domain** ([docs](https://cloud.google.com/storage/docs/static-website#tip-api)). 

In our case, we make preview builds of a site for each PR, and host them in a GCS bucket. But because of this issue, the Swift docs don't work: https://storage.googleapis.com/foxglove-mcap-dev-preview/cfcd27a5da7b4985342566570186f3aeab7bbe8c/docs/swift/documentation/mcap/

Compare this to the final hosted site, where it works: https://mcap.dev/docs/swift/documentation/mcap/

In short, I think the option to automatically add trailing `/` or even trailing `/index.html` should be configurable, because it's impossible for this project to anticipate all the different hosting solutions that developers will be using.
